### PR TITLE
Internal 133 frontend refactor

### PR DESCRIPTION
**Pull request by andrea-grapl**
Which issue does this PR correspond to?
grapl-security/issue-tracker#133

What changes does this PR make to Grapl? Why?
This PR doesn't make a functional changes to Grapl, but it does contain a reorganized file structure for src/js in addition to a "component map" which gives a brief overview of what each component is doing. This was created manually, but if it's useful, we'll eventually make a program to automate this process.

Notable Changes:
- Types are now in a folder
- Each component has its own folder containing its view, styles, and helper functions
- Each service has it's own folder containing API Calls
- Created fetch functions that created fetch calls in a function in services/fetch.tsx
- Addition of service map(see note above)
- Convert file paths to use absolute imports where possible (some deeply nested folders did not support absolute imports, kept relative for now)
- Added component map which shows the file structure for the front-end, in addition to a brief description about what each file does.

How were these changes tested?
Locally and manually.

### CODE REVIEW

File: src/js/engagement_view/src/components/engagementView/dynamicEngagementViewFeatures/utils/toggleNodeTable.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                        () => { toggle(toggled => !toggled) }
+                    }> 	
+                    <ExpandMoreIcon className={classes.expand}/> 
+                </Button>
+            </div>
+
+            <div className="nodeToggle">
+                {
+                    toggled && curNode && 
+                        <>
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/engagementView/dynamicEngagementViewFeatures/utils/toggleNodeTable.tsx:37]: these empty tags are new to me, what's the purpose? does it just say "hey we're doing jsx now"?

File: src/js/engagement_view/src/components/reusableComponents/graplHeader/GraplHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import Typography from "@material-ui/core/Typography";
+import Home from '@material-ui/icons/Home';
+
+import {graplHeaderStyles} from './styles';
+import {GraplHeaderProps} from 'types/GraplHeaderTypes'; 
+
+const useStyles = graplHeaderStyles; 
+
+const GraplHeader = (
+    {displayBtn}: GraplHeaderProps) => {
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/components/reusableComponents/graplHeader/GraplHeader.tsx:15]: sort of odd way to do function signature (with the leading parens on a diff line)

File: src/js/engagement_view/src/services/sagemakerNotebook/notebookService.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import  { apiFetch } from "../fetch";
+import DEV_API_EDGES from "../constants";
+
+export const getNotebookUrl = async (): Promise<string> => {
+  return apiFetch(`${DEV_API_EDGES.auth}/getNotebook`, "post")
+  .then(
+    (result) => result.success.notebook_url
+  );
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/services/sagemakerNotebook/notebookService.ts:8]: personally i prefer awaits over .then, but whatever :)

File: src/js/engagement_view/src/services/uploadPlugin/deletePlugin.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import DEV_API_EDGES from '../constants';
+import {apiFetchPostRequest} from '../fetch';
+
+export const deletePlugin = async ( pluginName: string ): Promise <boolean> => {
+    const body = JSON.stringify( {plugins_to_delete: [pluginName]} );
+    
+    try{ 
+        const response = await apiFetchPostRequest(`${DEV_API_EDGES.modelPluginEdge}/deleteModelPlugin`, "post", body);
```
Comment (wimax-grapl) [inline src/js/engagement_view/src/services/uploadPlugin/deletePlugin.tsx:8]: it's sort of odd to me that every call of apiFetchPostRequest requires you specify "post" manually - might be better named `apiFetchWithBody` or something?

**Review by wimax-grapl — APPROVED** [review]
overall looks good

File: src/js/engagement_view/src/components/engagementView/dynamicEngagementViewFeatures/utils/toggleNodeTable.tsx (hunk 1 above)
Comment (ajm-grapl) [reply src/js/engagement_view/src/components/engagementView/dynamicEngagementViewFeatures/utils/toggleNodeTable.tsx:37]: ya, they're like delimiters for inline HTML:  https://reactjs.org/docs/fragments.html#short-syntax

**Review by ajm-grapl — APPROVED** [review]

File: src/js/engagement_view/src/components/engagementView/dynamicEngagementViewFeatures/utils/toggleNodeTable.tsx (hunk 1 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/engagementView/dynamicEngagementViewFeatures/utils/toggleNodeTable.tsx:37]: This is a bad habit of mine, JSX requires parent elements, so I usually start with <></> but don't always need them. 

Removed. Nice Catch, thanks!

File: src/js/engagement_view/src/components/reusableComponents/graplHeader/GraplHeader.tsx (hunk 2 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/components/reusableComponents/graplHeader/GraplHeader.tsx:15]: Agreed and resolved.

File: src/js/engagement_view/src/services/uploadPlugin/deletePlugin.tsx (hunk 4 above)
Comment (andrea-grapl) [reply src/js/engagement_view/src/services/uploadPlugin/deletePlugin.tsx]: Agreed & updated.

**Pull request merged**